test/spec/token.js

```diff
@@ -0,0 +1,40 @@
+define(function(require) {
+  var OktaAuth = require('OktaAuth');
+  var Q = require('q');
+  var tokens = require('../util/tokens');
+
+  describe('token.decode', function () {
+
+    function setup() {
+      return Q.resolve(new OktaAuth({
+        url: 'http://example.okta.com'
+      }));
+    }
+
+    it('correctly decodes a token', function (done) {
+      return setup()
+      .then(function (oa) {
+        var decodedToken = oa.token.decode(tokens.unicodeToken);
+        expect(decodedToken).toDeepEqual(tokens.unicodeDecoded);
+        done();
+      });
+    });
+
+    it('throws an error for a malformed token', function (done) {
+      return setup()
+      .then(function (oa) {
+        return oa.token.decode('malformedToken');
+      })
+      .then(function (res) {
```

Comment:
is there something like a fail() in jasmine that will error out if it does hit this?